### PR TITLE
WIP: [examples] nodejs SDK example with nx angular nest template

### PR DESCRIPTION
**Pull request by benjaminstrasser**

### CODE REVIEW

**Comment by benjaminstrasser** [comment]
The existing example(#3738) uses a non-public repository as a source for the SDK, which seems to differ from the current one in the main branch.
Is #3719 and #3711 a good starting point for what will eventually be released?

I think it is important to see how the SDK behaves inside of a "production" project, where other tsconfig and package.json files can interfere to make sure it is useable there. Probably will be fine imo.

**Comment by slumbering** [comment]
> 



> The existing example(#3738) uses a non-public repository as a source for the SDK, which seems to differ from the current one in the main branch. Is #3719 and #3711 a good starting point for what will eventually be released?

The #3738 is now public

> I think it is important to see how the SDK behaves inside of a "production" project, where other tsconfig and package.json files can interfere to make sure it is useable there. Probably will be fine imo.

Thanks @benjaminstrasser for your contriibution. Very much appreciated.
Keep in mind that this SDK has not been merged yet but will be very soon. I'm currently working on another simple example with a React app which could be definitely a good starting point and will be available very soon in our docs. I'm really looking forward to see your use case as we need more of those.

File: examples/nx-example/README.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 This repository was automatically created by running `npx create-nx-workspace@15.0.13 nx-example --preset angular-nest --nxCloud false --appName nx-example` and choosing SCSS as a default stylesheet.
 
-## Steps to follow
+## Steps to setup this repo yourself
+Add the dagger dev dependency by running
+`$ npm i -D @dagger.io/nodejs-sdk`
```
Comment (slumbering) [inline examples/nx-example/README.md:7]: Update:
the package name will be publish as `@dagger.io/dagger`

Comment (slumbering) [reply examples/nx-example/README.md:7]: https://github.com/dagger/dagger/blob/61fce278f264820522aa061098b33a4231eac972/sdk/nodejs/package.json#L2

**Comment by mircubed** [comment]
@benjaminstrasser The SDK is already released and we have the documentation ready to go as well. We'd love your feedback on it before we launch! DM me on Discord (MirandaCarter#3268), so I can send you all the links, etc.

**Comment by benjaminstrasser** [comment]
@mircubed
Collecting all feedback here #3872 
will probably get a chance to look at it after work.

**Comment by slumbering** [comment]
@benjaminstrasser do you need anything for this PR to be reviewed ?

**Comment by benjaminstrasser** [comment]
@gerhard 
Didn't get to finish this, but made some progress that is not pushed yet. Will probably get around to finish this up this week.

**Comment by gerhard** [comment]
Great, looking forward to having this one merged! 💪

**Comment by benjaminstrasser** [comment]
Nx had an update that changed some things including adding vite, which I would love to try and which would make it lots easier to actually get around lots of ESM issues I hope. I am still on this just not my prio right now.

@gerhard 
In addition to the code, would you guys prefer a guide as well?
I had some major issues getting this to work and lots of ESM problems, which I assume many people might have.

**Comment by slumbering** [comment]
> 



> Nx had an update that changed some things including adding vite, which I would love to try and which would make it lots easier to actually get around lots of ESM issues I hope. I am still on this just not my prio right now.
> 
> @gerhard In addition to the code, would you guys prefer a guide as well? I had some major issues getting this to work and lots of ESM problems, which I assume many people might have.

That would be the icing on the cake :)
If you need any help let me know @benjaminstrasser, we can work together on that
Thanks

**Comment by slumbering** [comment]
@benjaminstrasser ! feel free to reopen this PR if needed. Thanks

**Pull request closed without merging**